source/Ship.cpp

```diff
@@ -1768,23 +1768,13 @@ int Ship::TakeDamage(const Projectile &projectile, bool isBlast)
 	bool wasDisabled = IsDisabled();
```

Comment:
You should probably use of these variables that are declared at the start of the function rather than using repeated calls to the weapon attributes (which I did in my example code just for the sake of clarity, because it was out of context):

```C++
	double shieldDamage = weapon.ShieldDamage();
	double hullDamage = weapon.HullDamage();
	double hitForce = weapon.HitForce();
	double heatDamage = weapon.HeatDamage();
	double ionDamage = weapon.IonDamage();
```